app/controllers/api_controller.rb

```diff
@@ -25,4 +25,12 @@ def list_find
       list = user.lists.first
     end
   end
+
+  def item_find
+    authenticate_or_request_with_http_basic do |email|
```

Comment:
Also, look into OAuth if you want a challenge instead of authenticating with basic_auth. It is pretty common now a days.